barretenberg/cpp/src/barretenberg/commitment_schemes/shplonk/shplemini_verifier.hpp

```diff
@@ -0,0 +1,319 @@
+\f[
+\#\text{claimed_evaluations} + \text{log_circuit_size} + 2
+\f]
+
+The ouput triple is either fed to  the corresponding \ref  bb::KZG< Curve_ >::reduce_verify_shplemini_accumulator
+"KZG method" or \ref  bb::IPA< Curve_ >::reduce_verify_shplemini_accumulator "IPA method". In the case of KZG, we reduce
+\f$ 6 \f$ batch_mul calls needed for the verification of the multivariate evaluation claims to the single batch_mul
+described above. In the case of IPA, the total number of batch_mul calls needed to verify the multivariate evaluation
+claims is reduced by \f$ 5 \f$.
+
+
+\todo Eliminate group operations spent on Shifted Polynomials. See <a
+href="https://github.com/AztecProtocol/barretenberg/issues/1084">the related issue</a>.
+*/
+template <typename Curve> class ShpleminiVerifier_ {
+    using Fr = typename Curve::ScalarField;
+    using GroupElement = typename Curve::Element;
+    using Commitment = typename Curve::AffineElement;
+    using VK = VerifierCommitmentKey<Curve>;
+    using ShplonkVerifier = ShplonkVerifier_<Curve>;
+    using GeminiVerifier = GeminiVerifier_<Curve>;
+
+  public:
+    template <typename Transcript>
+    static ShpleminiAccumulator<Curve> accumulate_batch_mul_arguments(Fr log_N,
+                                                                      RefSpan<Commitment> unshifted_commitments,
+                                                                      RefSpan<Commitment> shifted_commitments,
+                                                                      RefSpan<Fr> claimed_evaluations,
+                                                                      const std::vector<Fr>& multivariate_challenge,
+                                                                      const Commitment& g1_identity,
+                                                                      std::shared_ptr<Transcript>& transcript)
+    {
+        size_t log_circuit_size{ 0 };
+        if constexpr (Curve::is_stdlib_type) {
+            log_circuit_size = static_cast<uint32_t>(log_N.get_value());
+        } else {
+            log_circuit_size = static_cast<uint32_t>(log_N);
+        }
+        /// Get the challenge \f$\rho\f$ to batch commitments to multilinear polynomials and their shifts
+        const Fr multivariate_batching_challenge = transcript->template get_challenge<Fr>("rho");
+        /// Get Gemini commitments \f$(\text{com}(A_1), \text{com}(A_2), \ldots , \text{com}(A_{d-1}))\f$
+        std::vector<Commitment> gemini_commitments =
+            GeminiVerifier::get_gemini_commitments(log_circuit_size, transcript);
+        /// Get Gemini evaluation challenge for \f$ A_i \f$, \f$i = 0,\ldots, d-1\f$
+        const Fr gemini_evaluation_challenge = transcript->template get_challenge<Fr>("Gemini:r");
+        /// Get evaluations \f$ \left( A_0(-r), A_1(-r^2), ... , A_{d-1}(-r^{2^{d-1}}) \right) \f$
+        std::vector<Fr> gemini_evaluations(log_circuit_size);
```

Comment:
Why allocate and then overwrite? We should just coalesce to
```
std::vector<Fr> gemini_evaluations = GeminiVerifier_<Curve>::get_gemini_evaluations(log_circuit_size, transcript);
```
unless I'm missing something.

Reply:
Also this would enable it to be `const`. When `const` is missing I wonder if/why something is mutated. In this case, this vector is passed twice via const arguments and then accessed, so I think it should be const.